dart/lib/src/utils/http_sanitizer.dart

```diff
@@ -0,0 +1,82 @@
+import '../protocol.dart';
+import 'url_details.dart';
+
+class HttpSanitizer {
+  static final RegExp _authRegExp = RegExp("(.+://)(.*@)(.*)");
+
+  /// Parse and sanitize url data for sentry.io
+  static UrlDetails? sanitizeUrl(String? url) {
+    if (url == null) {
+      return null;
+    }
+
+    final queryIndex = url.indexOf('?');
+    final fragmentIndex = url.indexOf('#');
+
+    if (queryIndex > -1 && fragmentIndex > -1 && fragmentIndex < queryIndex) {
+      // url considered malformed because of fragment position
+      return UrlDetails();
+    } else {
+      final uri = Uri.parse(url);
+      final urlWithAuthRemoved = _urlWithAuthRemoved(uri._url());
+      return UrlDetails(
+          url: urlWithAuthRemoved.isEmpty ? null : urlWithAuthRemoved,
+          query: uri.query.isEmpty ? null : uri.query,
+          fragment: uri.fragment.isEmpty ? null : uri.fragment);
+    }
+  }
+
+  static Map<String, String>? sanitizedHeaders(Map<String, String>? headers) {
+    var mutableHeaders =
+        headers != null ? Map<String, String>.from(headers) : null;
```

Comment:
On Java individual cookies are filtered out, right? In this PR we remove the `Cookie` field completely instead of modifying its content. What should we do here?

Reply:
Not for all integrations. There doesn't seem to be alignment (yet). Some never send, some only send if `sendDefaultPii=true`. If we decide to send cookies they should be filtered to not contain tokens etc. (regardless of `sendDefaultPii`). Will try to get an answer.

Reply:
@denrase let's not add the headers that should be filtered out.
We can always add it later if needed.

Reply:
Right now we filter out all `Cookie` headers, that should suffice for now?

Reply:
yes